### PR TITLE
Fix bank cards logic and UI

### PR DESCRIPTION
**Pull request by BackSH00TER**
**Describe Pull request**
This PR adds the functionality needed for the bank cards section of qb-banking to work properly. Some of this functionality already existed in an incomplete state, this PR ties it all together. With these changes a player will now be able to view their active card in the banking screen. This will allow them to update their pin, lock their card, or request a new card.

- Added `bank_cards` table to the SQL file
- Added helper functions for getBankCard, addNewBankCard, toggleBankCardLock that update the database table.
- The logic assumes that a person can only ever have one active bank card. When the user requests a card for the first time, the new card is generated and then stored in the DB for the given citizenId. If a user requests a new card, the same function is used and it overwrites the database entry with the data for the new card. Note, this does not delete the actual item from the players inventory, so people could still have inactive bank cards in their inventory, only one of which is the active one.
- Additionally updated the request a new card UI to re-use the existing UI pieces for first time requesting a new card, adding a layer of consistency to the UI.
- Updated the UI for set pin to have a dark background instead of a light background

Notes:
- addBankCard makes uses of the SQL `REPLACE` and uses the citizenid as the primary key. The use of REPLACE will act just like INSERT if there are no results that match on the citizenid key. If there are existing results, it will replace the item with the new data
- Deleted `requestNewCard` and `qb-banking:createNewCard` as I am now re-using the same flow as when you get a card for the first time so these functions are now obsolete.

Screenshots:
Card management screen w/ Create a new pin (updated the background of the block from white to the same as other elements)
image

Card management screen - card locked
image

Requesting a new card and reusing the get a new card flow
fivem-request-new-card

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? [yes]
- Does your code fit the style guidelines? [yes]
- Does your PR fit the contribution guidelines? [yes]

### CODE REVIEW

File: server/main.lua [hunk 1]
```diff
@@ -204,16 +190,47 @@ local function format_int(number)
...
+local function getBankCard(cid)
+    local bankCard = MySQL.Sync.fetchAll('SELECT * FROM bank_cards WHERE citizenid = ? ORDER BY record_id DESC LIMIT 1', { cid })
+    return bankCard[1]
+end
+
+-- Adds a new bank card to the database, replaces existing card if it exists
+local function addNewBankCard(citizenid, cardNumber, cardPin, cardActive, cardLocked, cardType)
+    -- The use of REPLACE will act just like INSERT if there are no results that match on the citizenid key
+    -- If there are existing results, it will replace the item with the new data
+    MySQL.Async.insert('REPLACE INTO bank_cards (`citizenid`, `cardNumber`, `cardPin`, `cardActive`, `cardLocked`, `cardType`) VALUES (?, ?, ?, ?, ?, ?)', {
```
Comment (GhzGarage) [inline server/main.lua:203]: Change to MySQL.insert

File: server/main.lua [hunk 2]
```diff
@@ -204,16 +190,47 @@ local function format_int(number)
...
+        cardActive,
+        0, -- cardLocked
+        cardType
+    }, function(result)
+    end)
+end
+
+-- Toggle the lock status of a bank card
+local function toggleBankCardLock(cid, lockStatus)
+    MySQL.Async.execute('UPDATE bank_cards SET cardLocked = ? WHERE citizenid = ?', { lockStatus, cid},
```
Comment (GhzGarage) [inline server/main.lua:216]: Change to MySQL.update

File: server/main.lua [hunk 3]
```diff
@@ -298,15 +320,23 @@ RegisterNetEvent('qb-banking:doQuickWithdraw', function(amount, _)
...
+RegisterNetEvent('qb-banking:updatePin', function(currentBankCard, newPin)
+    if newPin ~= nil then
         local src = source
         local xPlayer = QBCore.Functions.GetPlayer(src)
         while xPlayer == nil do Wait(0) end
 
-        --   _char:Bank().UpdateDebitCardPin(pin)
-        TriggerClientEvent('qb-banking:openBankScreen', src)
-        TriggerClientEvent('qb-banking:successAlert', src, Lang:t('success.updated_pin'))
+        MySQL.Async.execute('UPDATE bank_cards SET cardPin = ? WHERE record_id = ?', {
```
Comment (GhzGarage) [inline server/main.lua:329]: Change to MySQL.update

File: server/main.lua [hunk 4]
```diff
@@ -204,16 +190,47 @@ local function format_int(number)
...
+    -- The use of REPLACE will act just like INSERT if there are no results that match on the citizenid key
+    -- If there are existing results, it will replace the item with the new data
+    MySQL.Async.insert('REPLACE INTO bank_cards (`citizenid`, `cardNumber`, `cardPin`, `cardActive`, `cardLocked`, `cardType`) VALUES (?, ?, ?, ?, ?, ?)', {
+        citizenid,
+        cardNumber,
+        cardPin,
+        cardActive,
+        cardLocked,
+        cardType
+    }, function(_)
```
Comment (BerkieBb) [inline server/main.lua:210]: This function can be removed, it is unnecessary when it doesn't execute any code

File: server/main.lua [hunk 5]
```diff
@@ -204,16 +190,47 @@ local function format_int(number)
...
+    end)
+end
+
+-- Toggle the lock status of a bank card
+local function toggleBankCardLock(cid, lockStatus)
+    MySQL.Async.execute('UPDATE bank_cards SET cardLocked = ? WHERE citizenid = ?', { lockStatus, cid},
+    function(_)
```
Comment (BerkieBb) [inline server/main.lua:217]: This function can be removed as well

**Review by BerkieBb — CHANGES_REQUESTED** [review]

File: server/main.lua (hunk 1 above)
Comment (BackSH00TER) [reply server/main.lua]: Fixed

File: server/main.lua (hunk 4 above)
Comment (BackSH00TER) [reply server/main.lua]: Fixed

File: server/main.lua [hunk 6]
```diff
@@ -204,16 +190,44 @@ local function format_int(number)
     return minus .. int:reverse():gsub("^,", "") .. fraction
 end
 
+-- Get all bank cards for the current player
+local function getBankCard(cid)
+    local bankCard = MySQL.Sync.fetchAll('SELECT * FROM bank_cards WHERE citizenid = ? ORDER BY record_id DESC LIMIT 1', { cid })
```
Comment (BerkieBb) [inline server/main.lua:195]: Use `MySQL.query.await` here

**Review by BerkieBb — CHANGES_REQUESTED** [review]

File: server/main.lua (hunk 6 above)
Comment (BackSH00TER) [reply server/main.lua]: Thanks, @BerkieBb updated to use MySQL.query.await

**Review by BerkieBb — APPROVED** [review]

**Review by iktray — APPROVED** [review]

**Comment by BackSH00TER** [comment]
Hi @BerkieBb and @GhzGarage apologies for the ping, is there anything you'd like me to change for this PR to be merged?

**Pull request merged**